soh/soh/Enhancements/enemyrandomizer.cpp

```diff
@@ -533,3 +538,45 @@ bool IsEnemyAllowedToSpawn(int16_t sceneNum, int8_t roomNum, EnemyEntry enemy) {
             return 1;
     }
 }
+
+void RegisterEnemyRandomizer() {
+    // prevent dark link from triggering a voidout
+    COND_VB_SHOULD(VB_TRIGGER_VOIDOUT, CVAR_ENEMY_RANDOMIZER_VALUE != CVAR_ENEMY_RANDOMIZER_DEFAULT, {
+        Actor* actor = va_arg(args, Actor*);
+
+        if (actor->category != ACTORCAT_PLAYER) {
+            *should = false;
+            Actor_Kill(actor);
+        }
+    });
+
+    // prevent dark link dealing fall damage to the player
+    COND_VB_SHOULD(VB_RECIEVE_FALL_DAMAGE, CVAR_ENEMY_RANDOMIZER_VALUE != CVAR_ENEMY_RANDOMIZER_DEFAULT, {
+        Actor* actor = va_arg(args, Actor*);
+
+        if (actor->category != ACTORCAT_PLAYER) {
+            *should = false;
+        }
+    });
+
+    // prevent dark link from interfering with HESS/recoil/etc when at more than 100 away from him
+    COND_VB_SHOULD(VB_TORCH2_HANDLE_CLANKING, CVAR_ENEMY_RANDOMIZER_VALUE != CVAR_ENEMY_RANDOMIZER_DEFAULT, {
+        Actor* darkLink = va_arg(args, Actor*);
+
+        if (darkLink->xzDistToPlayer > 100.0f) {
```

Comment:
what happens if this condition is same category check as those above? seems less hackish